src/main/java/chadbod/ChadBod.java

```diff
@@ -0,0 +1,176 @@
+                case BYE:
+                    ui.showFarewell();
+                    shouldExit = true;
+                    break;
+                case LIST:
+                    ui.printTasks(tasks);
+                    break;
+                case MARK:
+                    int markTaskNumber = getTaskNumber(details);
+                    Task markedTask = tasks.getTask(markTaskNumber);
+                    markedTask.markDone();
+                    ui.printStatusUpdate(true, markedTask);
+                    storage.saveTasks(tasks);
+                    break;
+                case UNMARK:
+                    int unmarkTaskNumber = getTaskNumber(details);
+                    Task unmarkedTask = tasks.getTask(unmarkTaskNumber);
+                    unmarkedTask.markUndone();
+                    ui.printStatusUpdate(false, unmarkedTask);
+                    storage.saveTasks(tasks);
+                    break;
+                case TODO:
+                    if (details.isEmpty()) {
+                        throw new InvalidTaskException("Description of todo cannot be empty.");
+                    }
+                    Todo newTodo = new Todo(details);
+                    tasks.addTask(newTodo);
+                    ui.printTaskAddedMessage(newTodo, tasks.getTaskCount());
+                    storage.saveTasks(tasks);
+                    break;
+                case DEADLINE:
+                    Deadline newDeadline = createDeadline(details);
+                    tasks.addTask(newDeadline);
+                    ui.printTaskAddedMessage(newDeadline, tasks.getTaskCount());
+                    storage.saveTasks(tasks);
+                    break;
+                case EVENT:
+                    Event newEvent = createEvent(details);
+                    tasks.addTask(newEvent);
+                    ui.printTaskAddedMessage(newEvent, tasks.getTaskCount());
+                    storage.saveTasks(tasks);
+                    break;
+                case DELETE:
+                    int taskNumber = getTaskNumber(details);
+                    Task removedTask = tasks.removeTask(taskNumber);
+                    ui.printTaskRemovedMessage(removedTask, tasks.getTaskCount());
+                    storage.saveTasks(tasks);
+                    break;
+                case FIND:
+                    TaskList matchingTasks = tasks.findTasksByKeyword(details);
+                    ui.printTasks(matchingTasks);
+                    break;
+                default:
+                    throw new InvalidInputException();
+                }
+            } catch (NumberFormatException e) {
+                ui.printErrorMessage("☹ OOPS!!! Invalid task index.");
+            } catch (ChadBodException e) {
+                ui.printErrorMessage(e.getMessage());
+            }
+        }
+    }
+
+    private int getTaskNumber(String details) throws NumberFormatException, TaskIndexOutOfBoundsException {
```

Comment:
Is there a better name for `details` to communicate what the input parameter is?

Reply:
Yes I agree, a more descriptive name for details would be good! Perhaps TaskDescription?